src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft. All rights reserved.
```

Comment:
This file has a bunch of places where you dereference a pointer (either by *ptr or ptr->) without doing a nullptr check. Please add ```assert(ptr != nullptr)``` on all of the functions that use pointers or add individual checks before dereferencing. 

The asserts would be preferable if the pointers are passed in by the framework itself and not from user code; then would could catch invalid usage from within the framework. If the user provides an object that gets passed directly to these functions, then a runtime check should occur with the correct failure code returned so we do not crash